### PR TITLE
HSD8-513 added color background option to text area paragraph

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Add color field to text area paragraph for background color

# Needed By (Date)
- soon

# Urgency
- low

# Steps to Test

1. checkout this branch
1. sync a site `blt drupal:sync --site=[sitename]`
1. create/edit a basic page
1. add a text area paragraph
1. enter some text or stuff and choose a background color from the options
1. verify the text area paragraph has the appropriate background colors.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5

### CODE REVIEW

**Comment by pookmish** [comment]
poke

File: config/default/core.entity_view_display.paragraph.hs_text_area.default.yml [hunk 1]
```diff
@@ -18,7 +21,12 @@ content:
     weight: 0
     label: hidden
     settings: {  }
-    third_party_settings: {  }
+    third_party_settings:
+      field_formatter_class:
+        class: 'color-[paragraph:field_hs_text_area_bg_color]'
```
Comment (sherakama) [inline config/default/core.entity_view_display.paragraph.hs_text_area.default.yml:26]: These are not the most semantic names but I see why you did this. I think for H&S this is fine but overall, these should have human-readable labels.

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/scss/hs_text_area.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
+@import 'colors', 'decanter-no-markup';
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/scss/hs_text_area.scss:3]: This should have charset declared at the top of the file.

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/scss/_colors.scss [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  }
+
+  a,
+  a:focus,
+  a:hover {
+    color: #dad7cb;
+  }
+}
+
+@mixin text-area-colors($color) {
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/scss/_colors.scss:193]: Mixins should have docblock style comments.

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/scss/_colors.scss [hunk 4]
```diff
@@ -0,0 +1,200 @@
+
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/scss/_colors.scss:2]: Missing charset declaration.

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/package.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "su_humsci_theme",
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/package.json]: Thi is not the theme. Please change package name.

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/package.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  "description": "Drupal 8 subtheme of Stanford Basic for the School of Humanities and Sciences",
+  "author": "Stanford Web Services",
+  "version": "1.0.0",
+  "license": "MIT",
+  "repository": "https://github.com/SU-HSDO/suhumsci",
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/package.json]: This is not the repo

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/package.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  "devDependencies": {
+    "bourbon": "^5.0.0",
+    "bourbon-neat": "^1.8.0",
+    "decanter": "^3.0.0",
+    "grunt": "^1.0.3",
+    "grunt-autoprefixer": "^3.0.3",
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/package.json:12]: We need to update our standard file, this is out of date. grunt-autoprefixer is out and autoprefixer is in.

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/hs_text_area.module [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function hs_text_area_display_paragraph_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+  if ($entity->bundle() == 'hs_text_area') {
+    $build['#attached']['library'][] = 'hs_text_area/hs_text_area';
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/hs_text_area.module:16]: Yay, libraries.

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/hs_text_area.libraries.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+
+hs_text_area:
+  version: VERSION
+  css:
+    base:
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/hs_text_area.libraries.yml]: This seems like a 'theme' layer change.

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/Gruntfile.js [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+
+  // This is where we tell Grunt we plan to use this plug-in.
+  grunt.loadNpmTasks('grunt-contrib-uglify');
+  grunt.loadNpmTasks('grunt-contrib-imagemin');
+  grunt.loadNpmTasks('grunt-svgmin');
+  grunt.loadNpmTasks('grunt-sass');
+  grunt.loadNpmTasks('grunt-contrib-watch');
+  grunt.loadNpmTasks('grunt-browser-sync');
+  grunt.loadNpmTasks('grunt-available-tasks');
+  grunt.loadNpmTasks('grunt-drush');
```
Comment (sherakama) [inline docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/Gruntfile.js]: This is not compatible with all local systems. Please remove drush.

**Review by sherakama — CHANGES_REQUESTED** [review]
Please see comments.

File: config/default/core.entity_view_display.paragraph.hs_text_area.default.yml (hunk 1 above)
Comment (pookmish) [reply config/default/core.entity_view_display.paragraph.hs_text_area.default.yml:26]: So the field that the token is coming from is actually a color field. And the data is stored as the value. It's a great module but that was my only dislike. Could add some altering if necessary in the future

File: docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/scss/hs_text_area.scss (hunk 2 above)
Comment (pookmish) [reply docroot/modules/humsci/hs_paragraph_types/modules/hs_text_area/scss/hs_text_area.scss:3]: This is what you get for having a back end developer do front end work 😁

**Comment by boznik** [comment]
This is working as expected. GTG by me when shea's changes are addressed

**Review by sherakama — APPROVED** [review]
GTG

**Pull request merged**